documentation/architecture.md

```diff
@@ -0,0 +1,247 @@
+# Clear Containers Architecture
+
+## Overview
+
+This is an architectural overview of Clear Containers, based on the 2.1 release.
+
+The [Clear Containers runtime (cc-oci-runtime)](https://github.com/01org/cc-oci-runtime)
+complies with the [OCI](https://github.com/opencontainers) specifications and thus
+work seamlessly with the [Docker Engine](https://www.docker.com/products/docker-engine)
+pluggable runtime architecture. In other words, one can transparently replace the
+[default Docker runtime (runc)](https://github.com/opencontainers/runc) with `cc-oci-runtime`.
+
+![Docker and Clear Containers](https://github.com/01org/cc-oci-runtime/blob/topic/documentation/documentation/docker-cc.png)
+
+`cc-oci-runtime` creates a QEMU/KVM virtual machine for each container the Docker engine
+tries to create.
+
+The container process is then spawned by [hyperstart](https://github.com/hyperhq/hyperstart/),
+an agent running as a daemon on the guest operating system.
+Hyperstart opens 2 virtio serial interfaces (Control and I/O) on the guest, and QEMU exposes them
+as serial devices on the host. `cc-oci-runtime` uses the control device for sending container
+management commands to hyperstart while the I/O serial device is used to pass I/O streams (`stdout`,
+`stderr`, `stdin`) between the guest and the Docker Engine.
+
+For any given container, both the init process and all potentially executed commands within that
+container, together with their related I/O streams, need to go through 2 virtio serial interfaces
+exported by QEMU. The [Clear Containers proxy (`cc-proxy`)](https://github.com/01org/cc-oci-runtime/tree/master/proxy)
+multiplexes and demultiplexes those commands and streams for all container virtual machines.
+There is one single `cc-proxy` running instance per Clear Containers host.
```

Comment:
I'd write this as, "There is only one `cc-proxy` instance per Clear Containers host."